dp/kni_pkt_handler.c

```diff
@@ -522,3 +542,63 @@ free_kni_ports(void) {
 		}
 	}
 }
+
+#ifdef USE_AF_PACKET
+#define VETH_TX_RETRIES_LIMIT		5
+/*------------------------------------------------------------------------------*/
+void
+kern_packet_ingress(int portid,
+		   struct rte_mbuf *pkts_burst[PKT_BURST_SZ],
+		   unsigned nb_rx)
+{
+	/* Send e.g. ARP packets to the kernel */
+	uint32_t pkt_indx = 0, retries = 0, i;
+	while (nb_rx && retries < VETH_TX_RETRIES_LIMIT) {
+		
+		uint16_t tx_cnt = rte_eth_tx_burst(portid + NUM_SPGW_PORTS,
+						   0, &pkts_burst[pkt_indx], nb_rx);
+		nb_rx -= tx_cnt;
+		pkt_indx += tx_cnt;
+		retries++;
+	}
+
+	/* free those mbufs that failed to get transmitted */
+	for (i = 0; i < nb_rx; i++, pkt_indx++) {
+		RTE_LOG_DP(ERR, DP, "Failed to transmit pkt from %s --> %s\n",
+			   (portid == S1U_PORT_ID) ? "S1U_PORT_ID" : "SGI_PORT_ID",
```

Comment:
We can reuse the array with names indexed by port? same as above

Reply:
Thanks. This simplified code.